include/particles/particle_bbar.tcc

```diff
@@ -0,0 +1,32 @@
+//! Construct a particle with id and coordinates
+template <unsigned Tdim>
+mpm::ParticleBbar<Tdim>::ParticleBbar(Index id, const VectorDim& coord)
+    : mpm::Particle<Tdim>(id, coord) {
+  this->initialise();
+  // Clear cell ptr
+  cell_ = nullptr;
+  // Nodes
+  nodes_.clear();
+  // Set material containers
+  this->initialise_material(1);
+  // Logger
+  std::string logger =
+      "particle_bbar" + std::to_string(Tdim) + "d::" + std::to_string(id);
+  console_ = std::make_unique<spdlog::logger>(logger, mpm::stdout_sink);
```

Comment:
Since this call the base class, you just need to override the console_

Reply:
Do you mean only Lines 12-15 are needed?